src/api-binder/report.ts

```diff
@@ -139,11 +139,26 @@ export async function startReporting() {
 	])) as StateReportOpts;
 
 	let reportPending = false;
-	const doReport = async () => {
+	let reportDeferred = false;
+	const doReport = async (): Promise<void> => {
+		// Send a report if no report is pending
 		if (!reportPending) {
 			reportPending = true;
-			await reportCurrentState(reportConfigs);
-			reportPending = false;
+			try {
+				// Report changes in the current state
+				await reportCurrentState(reportConfigs);
+			} finally {
+				reportPending = false;
+				// Check if a report was requested while this one was processing
+				if (reportDeferred) {
+					reportDeferred = false;
```

Comment:
I like to think about this variable as `shouldReportNewerState` and we move this assignment right after the `getCurrentForReport()`, we should be fine.

Reply:
went with `checkForNewerState` but not 100% on it yet.